group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -850,11 +850,16 @@ StreamsGroup getOrCreateStreamsGroup(
     ) {
         Group group = groups.get(groupId);
 
+        // Streams groups are inserted immediately into the `groups` map to allow soft-state
         if (group == null) {
-            return new StreamsGroup(logContext, snapshotRegistry, groupId, metrics);
+            StreamsGroup newGroup = new StreamsGroup(logContext, snapshotRegistry, groupId, metrics);
+            groups.put(groupId, newGroup);
```

Comment:
This is the fix, right? We add the new group directly into the soft state? (same below)